lte/cloud/go/services/subscriberdb/obsidian/handlers/handlers.go

```diff
@@ -538,6 +543,28 @@ func makeSubscriberStateHandler(desiredState string) echo.HandlerFunc {
 	}
 }
 
+func getStateExactIMSIPrefix(networkID string, typeFilter []string, keyFilter []string, keyPrefix *string, serdes serde.Registry) (state_types.StatesByID, error) {
+	// post-filter functionality to only return states with the exact IMSI prefix as specified
+	states, err := state.SearchStates(networkID, typeFilter, keyFilter, keyPrefix, serdes)
+	if err != nil {
+		return nil, err
+	}
+
+	if !funk.IsEmpty(keyPrefix) {
+		for stateID := range states {
+			matches := imsiKeyRe.FindStringSubmatch(stateID.DeviceID)
+			if len(matches) != imsiExpectedMatchCount {
+				glog.Errorf("state device ID %s did not match IMSI-prefixed regex", stateID.DeviceID)
```

Comment:
This can be induced by a misbehaving gateway. So it's not a "platform" error. So just do glog.Infof https://magma.github.io/magma/docs/next/contributing/contribute_conventions#go